pkg/cloudproduct/generic/generic.go

```diff
@@ -16,20 +16,62 @@
 package generic
 
 import (
+	"agones.dev/agones/pkg/apis"
 	agonesv1 "agones.dev/agones/pkg/apis/agones/v1"
 	"agones.dev/agones/pkg/client/informers/externalversions"
 	"agones.dev/agones/pkg/portallocator"
+	"agones.dev/agones/pkg/util/runtime"
+	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/informers"
 )
 
 // New returns a new generic cloud product
 //
 //nolint:revive // ignore the unexported return; implements ControllerHooksInterface
-func New() (*generic, agonesv1.APIHooks, error) { return &generic{}, nil, nil }
+func New() (*generic, error) { return &generic{}, nil }
```

Comment:
Is there a reason for the second return value, since there is never a case where this can return an error? I don't think `New()` is implementing an interface that requires this so maybe it can be simplified?

Then test code that used this could have the call `generic.New()` inlined instead of ignoring the second return value, e.g. in `pkg/gameservers/controller_test.go`.

Reply:
Originally I was being a bit stodgy about all of the cloudproduct interfaces having a `New()` function that could return an error since the ones that have meaningful auto-detection might error on the k8s client, but you're right that as implemented, there's no cause. Let me fix this up.

Reply:
Done